src/india_api/internal/service/server.py

```diff
@@ -133,3 +173,185 @@ class GetHealthResponse(BaseModel):
+
+
+@server.get(
+    "/{source}/{region}/forecast",
+    tags=["Forecast Routes"],
+    status_code=status.HTTP_200_OK,
+)
+def get_forecast_timeseries_route(
+    source: ValidSourceDependency,
+    region: str,
+    db: DBClientDependency,
+    auth: dict = Depends(auth),
+    # TODO: add auth scopes
+    forecast_horizon: ForecastHorizon = ForecastHorizon.day_ahead,
+    forecast_horizon_minutes: Optional[int] = None,
+) -> GetForecastGenerationResponse:
+    """Function for the forecast generation route.
+
+    Args:
+        source: The source of the forecast, this is current wind or solar.
+        region: The region to get the forecast for.
+        forecast_horizon: The time horizon to get the data for. Can be 'latest', 'horizon' or 'day ahead'
+        forecast_horizon_minutes: The number of minutes to get the forecast for. forecast_horizon must be 'horizon'
+    """
+    values: list[PredictedPower] = []
+
+    try:
+        if source == "wind":
+            values = db.get_predicted_wind_power_production_for_location(
+                location=region,
+                forecast_horizon=forecast_horizon,
+                forecast_horizon_minutes=forecast_horizon_minutes,
+            )
+        elif source == "solar":
+            values = db.get_predicted_solar_power_production_for_location(
+                location=region,
+                forecast_horizon=forecast_horizon,
+                forecast_horizon_minutes=forecast_horizon_minutes,
+            )
+    except Exception as e:
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Error getting {source} power production: {e}",
+        ) from e
+
+    return GetForecastGenerationResponse(
+        values=[y.to_timezone(tz=local_tz) for y in values],
+    )
+
+
+@server.get(
+    "/{source}/{region}/forecast/da-csv", tags=["Forecast Routes"], response_class=FileResponse
+)
+def get_forecast_da_csv(
+    source: ValidSourceDependency,
+    region: str,
+    db: DBClientDependency,
+    auth: dict = Depends(auth),
+):
+    """
+    Route to get the day ahead forecast as a CSV file.
+    """
+    if source == "wind":
+        values = db.get_predicted_wind_power_production_for_location(
+            location=region,
+            forecast_horizon=ForecastHorizon.day_ahead,
+        )
+    elif source == "solar":
+        values = db.get_predicted_solar_power_production_for_location(
+            location=region, forecast_horizon=ForecastHorizon.day_ahead
+        )
+    else:
+        raise Exception(f"Source {source} needs to be wind or solar")
+
+    # format to dataframe
+    df = format_csv(values)
+
+    # make file format
+    now_ist = pd.Timestamp.now(tz="Asia/Kolkata")
+    tomorrow_ist = df["Date [IST]"].iloc[0]
+    csv_file_path = f"{region}_{source}_da_{tomorrow_ist}.csv"
+
+    description = f"This file is made for {region} for {source} for {tomorrow_ist}, this was made at {now_ist}"
```

Comment:
Maybe something like
```description = f"Generated for {region} for {source} for {tomorrow_ist}, Forecast up-to-date as of {now_ist}"```

Reply:
thanks, ill update it, not exactly this, but
`Forecast for {region} for {source} for {tomorrow_ist}. The Forecast was created at {now_ist}`